src/modules/emotes/index.js

```diff
@@ -56,6 +56,14 @@ class EmotesModule {
 
     return null;
   }
+
+  get length() {
+    return this.emoteProviders.reduce((accumulator, {emotes}) => accumulator + emotes.size, 0);
```

Comment:
it would be great to avoid recomputing this for every retrieval. is there a way to mark the store as dirty and recompute it one time?